test/unit/plugins/rest_spec.js

```diff
@@ -797,18 +831,102 @@ describe("The Rest plugin", function () {
 				expect (failed, "concurrency").to.be.true;
 			});
 		});
+
+		describe("setting the status of an instance to a state not handled by update", function () {
+
+			var instanceID;
+			var responseCode;
+			var originalInstance;
+
+			before(function () {
+
+				return instances.createInstance()
+				.then(function (instance) {
+					instanceID = instance.id;
+					return instances.getInstance({ id : instance.id });
+				})
+				.then(function (instance) {
+					originalInstance = {
+						id       : instance.id,
+						ami      : instance.ami,
+						type     : instance.type,
+						uri      : instance.uri,
+						state    : instance.state,
+						revision : instance.revision
+					};
+
+					var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+					.payload({
+						ami      : instance.ami,
+						type     : instance.type,
+						uri      : instance.uri,
+						state    : "failed",
+						revision : instance.revision
+					});
+					return request.inject(server);
+				})
+				.then(function (response) {
+					responseCode = response.statusCode;
+				});
+			});
+
+			it("throws an error", function () {
+				return instances.getInstance({ id : instanceID })
+				.then(function (response) {
+					expect(responseCode).to.equal(400);
+					expect(response).to.deep.equal(originalInstance);
+				});
+			});
+		});
+
+		describe("when there is an AWS error", function () {
+
+			var terminateInstancesStub;
+			var instanceID;
+
+			before(function () {
+				terminateInstancesStub = Sinon.stub(awsAdapter, "terminateInstances").rejects(new Error("AWS Error"));
+				instances.createInstance()
+				.then(function (instance) {
+					instanceID = instance.id;
+					var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+					.payload({
+						ami      : instance.ami,
+						type     : instance.type,
+						uri      : instance.uri,
+						state    : "terminating",
+						revision : instance.revision
+					});
+					return request.inject(server);
+				});
+			});
+
+			after(function () {
+				terminateInstancesStub.restore();
+			});
+
+			it("returns the instance with state set to failed", function () {
```

Comment:
Since this is the test suite for the REST API we should probably perform some assertions on the HTTP response and not just the DB model.